workflows/mcmicro.nf

```diff
@@ -76,12 +76,55 @@ workflow MCMICRO {
     // // Run Background Correction
     // BACKSUB(ASHLAR.out.tif, ch_markers)
     //BACKSUB(ASHLAR.out.tif, [[id: "backsub"], params.marker_sheet])
+    /*
+    if (params.backsub) {
+        BACKSUB(ASHLAR.out.tif, [[id:"$ASHLAR.out.tif[0]['id']"], params.marker_sheet])
+        ch_segmentation_input = BACKSUB.out.backsub_tif
+        ch_versions = ch_versions.mix(BACKSUB.out.versions)
+    } else {
+    */
+        ch_segmentation_input = ASHLAR.out.tif
+    /*
+    }
+    */
+
     //ch_versions = ch_versions.mix(BACKSUB.out.versions)
 
     // Run Segmentation
 
-    DEEPCELL_MESMER(ASHLAR.out.tif, [[:],[]])
-    ch_versions = ch_versions.mix(DEEPCELL_MESMER.out.versions)
+    params.segmentation_list = params.segmentation.split(',') as List
+    mesmer_out_mask = channel.empty()
+    if ("mesmer" in params.segmentation_list) {
+        DEEPCELL_MESMER(ch_segmentation_input, [[:],[]])
+        ch_versions = ch_versions.mix(DEEPCELL_MESMER.out.versions)
+        ch_segmentation_input
+            .join(DEEPCELL_MESMER.out.mask)
+            .dump(tag: 'MCQUANT in mesmer')
+            .multiMap { it ->
+                image: [it[0], it[1]]
+                mask: [[id: it[0]['id'], segmenter: 'mesmer'], it[2]]
+            }
+            .set { ch_mesmer_out }
+        mesmer_out_mask = ch_mesmer_out.mask
+        ch_versions = ch_versions.mix(DEEPCELL_MESMER.out.versions)
```

Comment:
You have these `ch_versions.mix` calls doubled up for both segmenters. Remove one of the duplicates.

Reply:
I was creating a second variable because `ch_mesmer_out.mask` didn't always exist because of the conditional.
I was able to fix it in the last commit after removing the conditional.

Reply:
The duplicates are still there. I was referring to writing `ch_versions = ch_versions.mix(DEEPCELL_MESMER.out.versions)` on 2 different lines, and the same for `CELLPOSE`.

Reply:
Oh, I see.  I've got that fixed now.  Looks like I had the same thing for the commented out backsub so I fixed that too.  Thanks for pointing that out.